scripts/inventory/consul_io.py

```diff
@@ -292,11 +292,11 @@ def is_service_available(self, node, service_name):
 
     def consul_get_kv_inmemory(self, key):
         result = filter(lambda x: x['Key'] == key, self.inmemory_kv)
-        return result.pop() if result else None
+        return list(result).pop() if list(result) else None
```

Comment:
You should convert the result to `list` only once and then use that twice, instead of converting it twice. (Same below.)